mlir/lib/Dialect/ArmNeon/Transforms/LowerContractionToSMMLAPattern.cpp

```diff
@@ -54,6 +54,8 @@ class LowerContractionToSMMLAPattern
     // Note: RHS is not transposed.
     mlir::VectorType lhsType = op.getLhsType();
     mlir::VectorType rhsType = op.getRhsType();
+    if (rhsType.getRank() < 2)
+      return failure();
```

Comment:
Since 0-D vectors can also reach this point, could you add another check making sure that if any of the operands doesn't have a rank, we also bail out (+ test for that)?

Reply:
I don't think `vector.contract` supports 0-D vectors. I couldn't create one without hitting errors: 
``` error: unexpected error: 'vector.contract' op operand #0 must be vector of any type values, but got 'vector<i32>'```

I did add "hasRank" check just in case.